packages/lib/src/vis/shared/InteractionMesh.tsx

```diff
@@ -1,15 +1,51 @@
-import type { MeshProps } from '@react-three/fiber';
 import { useThree } from '@react-three/fiber';
+import type { ReactNode } from 'react';
+import { useEffect } from 'react';
 
-function InteractionMesh(props: MeshProps) {
-  const { width, height } = useThree((state) => state.size);
+interface Props {
+  onPointerDown?: (evt: PointerEvent) => void;
+  onPointerMove?: (evt: PointerEvent) => void;
+  onPointerUp?: (evt: PointerEvent) => void;
+  onWheel?: (evt: WheelEvent) => void;
+  children?: ReactNode;
+}
+
+function InteractionMesh(props: Props) {
+  const { onPointerDown, onPointerMove, onPointerUp, onWheel, children } =
+    props;
+  const { domElement } = useThree((state) => state.gl);
+
+  useEffect(() => {
+    if (onPointerDown) {
+      domElement.addEventListener('pointerdown', onPointerDown);
+    }
+    if (onPointerMove) {
+      domElement.addEventListener('pointermove', onPointerMove);
+    }
+    if (onPointerUp) {
+      domElement.addEventListener('pointerup', onPointerUp);
+    }
+    if (onWheel) {
+      domElement.addEventListener('wheel', onWheel);
+    }
+
+    return () => {
+      if (onPointerDown) {
+        domElement.removeEventListener('pointerdown', onPointerDown);
+      }
+      if (onPointerMove) {
+        domElement.removeEventListener('pointermove', onPointerMove);
+      }
+      if (onPointerUp) {
+        domElement.removeEventListener('pointerup', onPointerUp);
+      }
+      if (onWheel) {
+        domElement.removeEventListener('wheel', onWheel);
+      }
+    };
+  });
 
-  return (
-    <mesh {...props}>
-      <meshBasicMaterial opacity={0} transparent />
-      <planeGeometry args={[width, height]} />
-    </mesh>
-  );
+  return <group>{children}</group>;
```

Comment:
Does it even need to render a group or `children`? We could just render it separately - e.g. 

```ts
<>
  <CanvasEvents {...eventHandlers} /> // suggested name (also thought of `EventsHelper`, `EventsProxy`, etc.)
  <PanMesh />
</>
```

Reply:
We should also reconsider renaming `PanMesh` and the like later, since they will no longer contain `mesh`es.